rudder-core/src/main/scala/com/normation/rudder/domain/logger/ComplianceDebugLogger.scala

```diff
@@ -57,6 +60,21 @@ object ComplianceDebugLogger extends Logger {
   override protected def _logger = LoggerFactory.getLogger("explain_compliance")
 
 
+  //we have one logger defined by node.
+  //they automatically expires after some time.
+
+  val nodeCache = CacheBuilder.newBuilder().maximumSize(1000).expireAfterWrite(10, TimeUnit.MINUTES).build(
```

Comment:
what happen if we have 2000 nodes, and want to report on only one ?
will we have to hope that this one will be in the first 1000 logger created, or i am not understanding how it works ?